crates/nu-cli/src/commands/sum.rs

```diff
@@ -51,13 +51,26 @@ impl WholeStreamCommand for Sum {
 
 fn sum(RunnableContext { mut input, .. }: RunnableContext) -> Result<OutputStream, ShellError> {
     let stream = async_stream! {
-        let mut values = input.drain_vec().await;
-
+        let mut values: Vec<Value> = input.drain_vec().await;
         let action = reducer_for(Reduce::Sum);
 
-        match action(Value::zero(), values) {
-            Ok(total) => yield ReturnSuccess::value(total),
-            Err(err) => yield Err(err),
+        if values.iter().all(|v| if let UntaggedValue::Primitive(_) = v.value {true} else {false}) {
+            let total = action(Value::zero(), values)?;
+            yield ReturnSuccess::value(total)
+        } else {
+            for value in values.into_iter() {
+                match value.value {
+                    UntaggedValue::Row(row_dict) => {
+                        let row_values = row_dict.entries.into_iter().map(|kvp| kvp.1).collect();
```

Comment:
If you wanted to make this a little bit clearer, you could use [`IndexMap#values`](https://docs.rs/indexmap/1.3.2/indexmap/map/struct.IndexMap.html#method.values):

```rust
let row_values = row_dict.entries.values().collect();
```

Reply:
I don't actually think that compiles since [`IndexMap#values`](https://docs.rs/indexmap/1.3.2/indexmap/map/struct.IndexMap.html#method.values) returns an iterator over `&Value` while the call to `action` on the following line takes a `Vec<Value>`.

Reply:
Ah, you're totally right. We'd have to update `action` to be a little more flexible, but no worries about that in this PR 🙂